File: app/src/main/java/info/horriblesubs/sher/api/horrible/response/Result.java
package info.horriblesubs.sher.api.horrible.response;

import java.io.Serializable;
import java.util.Date;
import java.util.List;

import info.horriblesubs.sher.api.horrible.model.DateParse;

public class Result<T> implements Serializable {

    public List<T> items;
    public String time;

    public Date getDate() {
        return DateParse.getNetworkDate(time + " +00:00");
    }

    public String getTime() {
        return DateParse.getNetworkDate(getDate());
    }

    public Result() {
    }
}

File: app/src/main/java/info/horriblesubs/sher/api/horrible/result/Result.kt
package info.horriblesubs.sher.api.horrible.result

import info.horriblesubs.sher.common.Formats
import info.horriblesubs.sher.common.timePassed
import info.horriblesubs.sher.common.toZonedDateTime
import java.time.ZonedDateTime

open class Result<T>(private val time: String?, val items: List<T>?) {
    internal val dateTime: ZonedDateTime?
        get() =  time?.toZonedDateTime()

    fun dateTime(formats: Formats): String? {
        return dateTime?.format(formats.formatter)
    }

    fun timePassed(): String {
        return dateTime?.timePassed(ZonedDateTime.now()) ?: "Never"
    }

    override fun toString(): String {
        return "time: $time\nitems: $items"
    }
}

fun <T> Result<T>?.isNull(): Boolean {
    return this == null || this.dateTime == null
}